src/java.xml/share/classes/com/sun/org/apache/xml/internal/utils/XMLReaderManager.java

```diff
@@ -50,12 +50,12 @@ public class XMLReaderManager {
     /**
      * Cache of XMLReader objects
      */
-    private ThreadLocal<ReaderWrapper> m_readers;
+    private volatile ThreadLocal<ReaderWrapper> m_readers;
```

Comment:
It's not very common to see a ThreadLocal as an instance field but here it's more like a TL per XMLReaderManager instance. Is there any reason why this can't be eagerly created and use a final field here?

Reply:
Yeah, a final static field would be good, though no harm was done in this case as the reader manager is a singleton. But final and then eagerly created would be better, get and release only need to reset states.